testing/unstable_snapshot.ts

```diff
@@ -0,0 +1,411 @@
+// Copyright 2018-2025 the Deno authors. MIT license.
+// @ts-nocheck Deno.lint namespace does not pass type checking in Deno 1.x
+
+import type { SnapshotOptions } from "./snapshot.ts";
+import { fromFileUrl } from "@std/path/from-file-url";
+import { parse } from "@std/path/parse";
+import { resolve } from "@std/path/resolve";
+import { toFileUrl } from "@std/path/to-file-url";
+import { AssertionError } from "@std/assert/assertion-error";
+import { equal } from "@std/assert/equal";
+import {
+  escapeStringForJs,
+  getIsUpdate,
+  getOptions,
+  getSnapshotNotMatchMessage,
+  LINT_SUPPORTED,
+  serialize,
+} from "./_snapshot_utils.ts";
+
+/**
+ * The options for {@linkcode assertInlineSnapshot}.
+ *
+ * @experimental **UNSTABLE**: New API, yet to be vetted.
+ */
+export interface InlineSnapshotOptions<T = unknown>
+  extends Pick<SnapshotOptions<T>, "mode" | "msg" | "serializer"> {
```

Comment:
How about dropping `mode` from `InlineSnapshotOptions`? That'll be inconsistent with `assertSnapshot`, but I don't think the users would ever specify `mode` via this option (People should specify `mode` only via CLI flag (`--update`)).